File: app/src/test/java/org/wikipedia/createaccount/CreateAccountInfoClientTest.java
package org.wikipedia.createaccount;

import org.junit.Test;
import org.wikipedia.dataclient.mwapi.MwQueryResponse;
import org.wikipedia.test.MockRetrofitTest;

import java.io.IOException;

import io.reactivex.rxjava3.core.Observable;

public class CreateAccountInfoClientTest extends MockRetrofitTest {

    @Test public void testRequestSuccess() throws Throwable {
        enqueueFromFile("create_account_info.json");
        getObservable().test().await()
                .assertComplete().assertNoErrors()
                .assertValue(response -> {
                    String token = response.getQuery().getCreateAccountToken();
                    String captchaId = response.getQuery().getCaptchaId();

                    return token.equals("5d78e6a823be0901eeae9f6486f752da59123760+\\")
                            && captchaId.equals("272460457");
                });
    }

    @Test public void testRequestResponse404() throws Throwable {
        enqueue404();
        getObservable().test().await()
                .assertError(Exception.class);
    }

    @Test public void testRequestResponseMalformed() throws Throwable {
        enqueueMalformed();
        getObservable().test().await()
                .assertError(IOException.class);
    }

    private Observable<MwQueryResponse> getObservable() {
        return getApiService().getAuthManagerInfo();
    }
}

File: app/src/test/java/org/wikipedia/createaccount/CreateAccountInfoClientTest.kt
package org.wikipedia.createaccount

import kotlinx.coroutines.runBlocking
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Test
import org.wikipedia.test.MockRetrofitTest

class CreateAccountInfoClientTest : MockRetrofitTest() {
    @Test
    @Throws(Throwable::class)
    fun testRequestSuccess() {
        enqueueFromFile("create_account_info.json")
        runBlocking {
            apiService.getAuthManagerInfo()
        }.run {
            assertEquals("5d78e6a823be0901eeae9f6486f752da59123760+\\", query!!.createAccountToken())
            assertEquals("272460457", query!!.captchaId())
        }
    }

    @Test
    @Throws(Throwable::class)
    fun testRequestResponse404() {
        enqueue404()
        runBlocking {
            try {
                apiService.getAuthManagerInfo()
            } catch (e: Exception) {
                assertNotNull(e)
            }
        }
    }

    @Test
    @Throws(Throwable::class)
    fun testRequestResponseMalformed() {
        enqueueMalformed()
        runBlocking {
            try {
                apiService.getAuthManagerInfo()
            } catch (e: Exception) {
                assertNotNull(e)
            }
        }
    }
}
